blog/docs/articles/consuming_s3_data_with_knative.md

```diff
@@ -0,0 +1,175 @@
+
+## Installation
+
+The [Installation](https://camel.apache.org/camel-k/next/installation/installation.html) from Apache Camel K offers a few choices, such as CLI, Kustomize, OLM or Helm. Example of Helm installation:
+
+```
+$ helm repo add camel-k https://apache.github.io/camel-k/charts/
+$ helm install my-camel-k camel-k/camel-k
+```
+
+Besides Camel K we also need to have Knative Eventing installed, as described in the [documentation](https://knative.dev/docs/install/yaml-install/eventing/install-eventing-with-yaml/){:target="_blank"}.
+
+## Creating a Knative Broker instance
+
+We are using a Knative Broker as the heart of our system, acting as an [Event Mesh](https://knative.dev/docs/eventing/event-mesh/){:target="_blank"} for both event producers and event consumers:
+
+```yaml
+apiVersion: eventing.knative.dev/v1
+kind: Broker
+metadata:
+  namespace: default
+  name: my-broker
+```
+
+Now event producers can send events to it and event consumers can receive events.
+
+## Using Kamelets as Event Sources for AWS S3
+
+In order to bind a Kamelet to a Knative component, like the above broker, we are using the `Pipe` API. A Pipe allows to declaratively move data from a system described by a Kamelet _towards_ a Knative destination **or** _from_ a Knative destination to another (external) system described by a Kamelet.
+
+Below is a `Pipe` that uses a ready-to-use `Kamelet`, a `aws-s3-source`
+
+```yaml
+apiVersion: camel.apache.org/v1
+kind: Pipe
+metadata:
+  name: aws-s3-source-pipe
+  annotations:
+    trait.camel.apache.org/mount.config: "secret:aws-s3-credentials"
+spec:
+  integration:
+    dependencies:
+    - "camel:cloudevents"
+  source:
+    ref:
+      kind: Kamelet
+      apiVersion: camel.apache.org/v1
+      name: aws-s3-source
+    properties:
+      bucketNameOrArn: "${aws.s3.bucketNameOrArn}"
+      accessKey: "${aws.s3.accessKey}"
+      secretKey: "${aws.s3.secretKey}"
+      region: "${aws.s3.region}"
+    dataTypes:
+      out:
+        scheme: aws2-s3
+        format: application-cloudevents
+  sink:
+    dataTypes:
+      in:
+        scheme: http
+        format: application-cloudevents
+    ref:
+      kind: Broker
+      apiVersion: eventing.knative.dev/v1
+      name: my-broker
+```
+
+
+The `aws-s3-source` Kamelet is referenced as the `source` of the `Pipe` and sends CNCF CloudEvents to the outbound `sink`, when there is activity in the referenced bucket. Here we use the Knative Broker, which accepts CloudEvents. Later we will connect a Knative Function to the Broker for receiving events from it.
+
+!!! note
+
+    The AWS S3 properties are stored in a secret, which is mounted into the `Pipe`, via the `trait.camel.apache.org/mount.config` annotation.
+
+
+## Creating a Knative Function as a consumer
+
+In order to consume messages from the Knative broker, using Knative Function, we need will create a simple Golang function with the following command:
```

Comment:
Maybe we can mention that it consumes cloudevents here (as opposed to http)? I think this would make the `-t cloudevents` make more sense on the next line